File: sem4/mpp/project-cs/networking/Response.cs
using Newtonsoft.Json;
using project_cs.domain;

namespace networking;

public enum ResponseType {
    OK,
    ERROR,
    CHILDREN_LIST,
    UPDATE_CHILDREN
}

public class Response {
    public ResponseType type { get; set; }
    public string errorMessage { get; set; }
    public List<ChildDto> children { get; set; }
}

File: sem4/mpp/project-java/networking/src/main/java/networking/Response.java
package networking;

import domain.dto.ChildDTO;
import java.io.Serializable;
import java.util.List;


public class Response {
	private ResponseType type;
	private String errorMessage;

	private List<ChildDTO> children;

	public Response() {
	}

	public ResponseType getType() {
		return type;
	}

	public void setType(ResponseType type) {
		this.type = type;
	}

	public String getErrorMessage() {
		return errorMessage;
	}

	public void setErrorMessage(String errorMessage) {
		this.errorMessage = errorMessage;
	}

	public List<ChildDTO> getChildren() {
		return children;
	}

	public void setChildren(List<ChildDTO> children) {
		this.children = children;
	}

	@Override
	public String toString() {
		return "Response{" +
				"type=" + type +
				", errorMessage='" + errorMessage + '\'' +
				", children=" + children +
				'}';
	}
}
